src/tuyaAPI31.cpp

```diff
@@ -29,7 +29,7 @@
 #include <iostream>
 #endif
 
-int tuyaAPI31::BuildTuyaMessage(unsigned char *buffer, const uint8_t command, const std::string &szPayload, const std::string &encryption_key = "")
+int tuyaAPI31::BuildTuyaMessage(unsigned char *buffer, const uint8_t command, const std::string &szPayload, const std::string &szEncryptionkey)
```

Comment:
#8 cleans up the API so that we don't have to pass the encryption key into BuildTuyaMessage every time, and it's set up once by an explicit NegotiateSession() call (and for the older protocols, setting the key is *all* the NegotiateSession() call does). Please let's not make other changes that conflict with that before merging it :)